### PR TITLE
add conda recipe for compaction

### PR DESCRIPTION
**Pull request by mcflugen**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by xhochy** [comment]
This can probably be noarch but you may need to declare the entrypoints: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#python-entry-points

File: recipes/compaction/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+source:
+  url: https://pypi.io/packages/source/c/compaction/compaction-{{ version }}.tar.gz
+  sha256: f13dfad27140295f850a1ea8a8b140d5ce483b3d3a4f2ca701203b7a9493ee87
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+  skip: True  # [py2k]
```
Comment (scopatz) [inline recipes/compaction/meta.yaml]: Please remove this line for noarch

**Comment by mcflugen** [comment]
@scopatz, @xhochy Thanks for your help. It is very much appreciated!

I think this should be ready to go now, I'm waiting for the tests to complete, but I think they may be stuck as they've been running for a while now.

**Comment by mariusvniekerk** [comment]
Restarted the test suite

**Comment by mcflugen** [comment]
My build stalled again. Is it my recipe?

**Comment by mcflugen** [comment]
Woohoo! The builds finally finished 🎉 .

@conda-forge/help-python This is now ready to go. Thanks!

File: recipes/compaction/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/c/compaction/compaction-{{ version }}.tar.gz
+  sha256: f13dfad27140295f850a1ea8a8b140d5ce483b3d3a4f2ca701203b7a9493ee87
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
```
Comment (marcelotrevisani) [inline recipes/compaction/meta.yaml]: It can be just
```suggestion
  script: "{{ PYTHON }} -m pip install . -vv"
```

Comment (mcflugen) [reply recipes/compaction/meta.yaml]: Thanks @marcelotrevisani. I've made this change.

**Pull request merged**